src/SdfEntityCreator.cc

```diff
@@ -814,20 +814,20 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Visual *_visual)
       {
         gzwarn << "Using an internal gazebo.material to parse "
           << scriptName << std::endl;
-        MaterialParser::MaterialValues parsed =
+        std::optional<MaterialParser::MaterialValues> parsed =
           this->dataPtr->materialParser.GetMaterialValues(scriptName);
 
-        if(!parsed.ambient.has_value()) {
+        if(!parsed.has_value()) {
           gzwarn << "Material " << scriptName <<
-            " not recognized, using default."<< std::endl;
+            " not recognized, using default." << std::endl;
         }
 
         visualMaterial.SetAmbient
```

Comment:
There should be an `else` here. Otherwise, we'd try to access `parsed` when it doesn't have a value.

Reply:
I think the value_or would provide a default in this case.

Reply:
The problem is with `parsed->`. It's not throwing an exception now since `MaterialParser::GetMaterialValues` currently always returns the optional with a value in it. Addressing my comment there would make it so a nullopt would be returned if a material is not found, in which case `parsed->` would throw an exception.